crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -439,15 +439,13 @@ impl Instruction {
                     } else if obj.is_zero() {
                         if let Some(location) = *location {
                             return Err(RuntimeError::new(
-                                RuntimeErrorKind::UnstructuredError {
-                                    message: "Constraint is always false".into(),
-                                },
+                                RuntimeErrorKind::ConstraintIsAlwaysFalse { spanless: false },
                                 Some(location),
                             ));
                         } else {
-                            return Err(RuntimeErrorKind::Spanless(
-                                "Constraint is always false".into(),
-                            )
+                            return Err(RuntimeErrorKind::ConstraintIsAlwaysFalse {
+                                spanless: true,
+                            }
```

Comment:
I think these two if cases can be combined now. Something like:
```rs
return Err(RuntimeError::new(
    RuntimeErrorKind::ConstraintIsAlwaysFalse { spanless: location.is_none() },
    *location,
});
```

Reply:
Although I'm not sure we need the "spanless" field at all anymore now that we have an optional location.